processor/worker.go

```diff
@@ -13,15 +13,14 @@ import (
 	"github.com/rudderlabs/rudder-server/utils/misc"
 )
 
-// newWorker creates a new worker
-func newWorker(ctx context.Context, partition string, h workerHandle) *worker {
+// newProcessorWorker creates a new worker
+func newProcessorWorker(partition string, h workerHandle) *worker {
 	w := &worker{
 		handle:    h,
 		logger:    h.logger().Child(partition),
 		partition: partition,
 	}
-	w.lifecycle.ctx, w.lifecycle.cancel = context.WithCancel(ctx)
-	w.channel.ping = make(chan struct{}, 1)
+	w.lifecycle.ctx, w.lifecycle.cancel = context.WithCancel(context.Background())
```

Comment:
Curious why did we stop propagating the context from parent

Reply:
For controlling the worker's lifecycle exclusively through `Stop`. If the parent context is canceled, the worker's stop will eventually be called by the pool